common/primitives/src/handles.rs

```diff
@@ -58,6 +63,31 @@ impl<BlockNumber> ClaimHandlePayload<BlockNumber> {
 	}
 }
 
+impl<BlockNumber> EIP712Encode for ClaimHandlePayload<BlockNumber>
+where
+	BlockNumber: Into<U256> + TryFrom<U256> + Copy,
+{
+	fn encode_eip_712(&self) -> Box<[u8]> {
+		lazy_static! {
+			// get prefix and domain separator
+			static ref PREFIX_DOMAIN_SEPARATOR: Box<[u8]> =
+				get_eip712_encoding_prefix("0xcccccccccccccccccccccccccccccccccccccccc");
```

Comment:
These are hardcoded for each payload and most probably would need to be replaced with precompile addresses when implemented.